libimage/filters.go

```diff
@@ -79,7 +79,7 @@ func (r *Runtime) compileImageFilters(ctx context.Context, options *ListImagesOp
 	var tree *layerTree
 	getTree := func() (*layerTree, error) {
 		if tree == nil {
-			t, err := r.layerTree(ctx, nil)
+			t, err := r.layerTree(nil, nil)
```

Comment:
See the call stack — ideally it seems to me that `ListImages` should create a `layerTree` and pass it down here, for both efficiency and to make the filter outcome and the list output consistent.

OTOH that would require restructuring the filter image compilation to allow answering the question “do we need a layer tree” before the “real” compilation happens. Or, maybe easier, have the outcome of the compilation return a “BTW I need a layer tree” value, and only pass the layer tree down into the filter _execution_ phase.